Fix fc_atomic_ptr_cmpexch on arm64 so that it does not crash.

Upstream fixed it in a different way that removed support for Tiger.
We will want to be careful to patch that back in.

https://github.com/freedesktop/fontconfig/commit/6def66164a36eed968aae872d76acfac3173d44a
--- src/fcatomic.h.orig	2016-12-01 21:22:19.000000000 -0600
+++ src/fcatomic.h	2020-09-04 00:25:00.000000000 -0500
@@ -83,7 +83,7 @@
 #if (MAC_OS_X_VERSION_MIN_REQUIRED > MAC_OS_X_VERSION_10_4 || __IPHONE_VERSION_MIN_REQUIRED >= 20100)
 #define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwapPtrBarrier ((void *) (O), (void *) (N), (void **) (P))
 #else
-#if __ppc64__ || __x86_64__
+#if __LP64__
 #define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwap64Barrier ((int64_t) (O), (int64_t) (N), (int64_t*) (P))
 #else
 #define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwap32Barrier ((int32_t) (O), (int32_t) (N), (int32_t*) (P))
